app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleState.kt

```diff
@@ -37,6 +37,11 @@ sealed class ContactSearchResult(val searchResultState: SearchResultState) {
 sealed class SearchResultState {
     object Initial : SearchResultState()
     object InProgress : SearchResultState()
-    data class Failure(val failureMessage: String? = null) : SearchResultState()
+
+    sealed class Failure : SearchResultState() {
+        data class GenericFailure(val failureMessage: String? = null) : Failure()
+        object InvalidQueryFailure : Failure()
+    }
+
```

Comment:
how about instead of making two Failure hierarchy types we do something like this : 

data class Failure(val message : String? = null) : SearchResultState()

then inside the VIewModel we can assign the message depending on the reason of the failure like : 


SearchResultState.Failure("Not found") or 
SearchResultState.Failure("No results")

Reply:
the this is I couldn't pass the text in the VM as either it need context to use getString 
or the   stringResource() can't pass it to the VM as well because it need to be inside a composable component 
in the Text text for example

Reply:
you could pass a R.string.something

Reply:
if you pass it in the ViewModel you can consume it later in the UI by using stringResource()